hack/docker.sh

```diff
@@ -26,11 +26,6 @@ docker-login() {
     local image=$1
     local registry=${image%%"/"*}
 
-    if grep -q "$registry" ~/.docker/config.json; then
```

Comment:
Does this cause problems with incorrect push permissions or why are we removing it? Curious because is came up in the context of https://github.com/elastic/cloud-on-k8s/issues/3906

Reply:
I couldn't pinpoint the exact cause, but, removing this check is the way to make the CI image build succeed. My theory is that we are getting a very short-lived Docker credential from Vault. Because it takes a while to build the CI image, by the time it is done the credential has already expired.

I couldn't see a reason for having this check in here. It's not a particularly expensive operation to do. Given that the build randomly fails because of it, I thought it would be better to remove it.

Reply:
Agreed!